angularSTEP/src/app/upload-recipe/upload-recipe.component.html

```diff
@@ -0,0 +1,84 @@
+<h3>Upload a Recipe</h3>
+
+<form [formGroup]="recipeForm" (submit)="onSubmit()">
+  <mat-form-field>
+    <mat-label>Recipe Name</mat-label>
+    <input matInput placeholder="Ex. Mom's tamales" formControlName="name" required>
+    <mat-error>This is required.</mat-error>
+  </mat-form-field>
+
+  <br><br>
+
+  <mat-form-field>
+    <mat-label>Recipe Difficulty</mat-label>
+    <mat-select formControlName="difficulty" required>
+      <mat-option value="Beginner">Beginner</mat-option>
+      <mat-option value="Intermediate">Intermediate</mat-option>
+      <mat-option value="Advanced">Advanced</mat-option>
+      <mat-option value="Professional">Professional</mat-option>
+    </mat-select>
+    <mat-error>This is required.</mat-error>
+  </mat-form-field>
+
+  <br><br>
+
+  <mat-form-field>
+    <mat-label>Short Description</mat-label>
+    <input matInput placeholder="Ex. Best Chicken tamales ever. NOT for vegetarians!" formControlName="description" required>
+    <mat-error>This is required.</mat-error>
```

Comment:
This is just a possible suggestion for a future commit, however, we might want a character limit on the Short Description. In a mat-form-field, I know there is a way to display the remaining characters I believe. For now, you may want to add Validator.maxLength to like 100? Then throw a separate error if that validator fails.

Reply:
Good idea! I'll do that now

Reply:
Note that this enforcement only takes place here in the browser form. I could still arbitrarily send a POST request to your backend server with a longer message and AFAIK there's nothing there preventing that from being accepted, so this will need to be augmented with backend validation as well - might be worth tracking with a GitHub issue?